docs/manual/must-call-checker.tex

```diff
@@ -0,0 +1,204 @@
+\htmlhr
+\chapterAndLabel{Must Call Checker}{must-call-checker}
+
+The Must Call Checker conservatively over-approximates
+the set of methods that an object should call before it is de-allocated.
+It is intended to be run as a subchecker of another checker: on its own, it does not
+enforce any rules other than subtyping.
```

Comment:
Does the MustCall Checker refine the type of an expression if one of the must-call methods have been called?

```
@MustCall("close") OutputStream o = ...;
o.close();
@MustCall({}) OutputStream p = o; // Does this type check?
```

Also, the MustCall checker doesn't issue an error if a must-call method isn't called right before it is de-allocated, right?

Reply:
The code example you gave does not typecheck; the assignment to `p` is forbidden.

> Also, the MustCall checker doesn't issue an error if a must-call method isn't called right before it is de-allocated, right?

No. The Must Call Checker never issues an error before an object is de-allocated, regardless of its must-call type. As the sentence says, it enforces no rules other than subtyping. That's also why the manual says "It is intended to be run as a subchecker of another checker" - this is one (relatively independent) component of a composite checker that will check that every must-call method is actually called before each object is de-allocated.

Reply:
After the very first sentence in this section, I would add something like:
"The checker does not enforce that the methods are called before objects are de-allocated, nor is the MustCall annotation refined after the a must-call method is called."

That would help clarify what it does.

Reply:
I re-worked that whole paragraph to be more explicit. It might be worth pointing out that this checker's manual section will appear with the other "subcheckers by design", such as the Value Checker and the MethodVal Checker, in the manual.